lib/usdUfe/ufe/UsdContextOps.cpp

```diff
@@ -0,0 +1,480 @@
+
+    // Subtract the set of what *is* loaded on the stage from the set of what
+    // *can* be loaded at or below this prim path. The resulting set will
+    // contain all paths that are loadable, but not currently loaded, at or
+    // below this prim path.
+    SdfPathSet unloadedSet;
+    std::set_difference(
+        loadableSet.cbegin(),
+        loadableSet.cend(),
+        stageLoadSet.cbegin(),
+        stageLoadSet.cend(),
+        std::inserter(unloadedSet, unloadedSet.end()));
+
+    if (!unloadedSet.empty()) {
+        // Loading without descendants is only meaningful for context ops when
+        // the current prim has an unloaded payload.
+        if (prim.HasPayload() && !prim.IsLoaded()) {
+            itemLabelPairs.emplace_back(std::make_pair(kUSDLoadItem, kUSDLoadLabel));
+        }
+
+        // We always add an item for loading with descendants when there are
+        // unloaded paths at or below the current prim, since we may be in one
+        // of the following situations:
+        // - The current prim has a payload that is unloaded, and we don't know
+        //   whether loading it will introduce more payloads in descendants, so
+        //   we offer the choice to also load those or not.
+        // - The current prim has a payload that is loaded, so there must be
+        //   paths below it that are still unloaded.
+        // - The current prim does not have a payload, so there must be paths
+        //   below it that are still unloaded.
+        itemLabelPairs.emplace_back(
+            std::make_pair(kUSDLoadWithDescendantsItem, kUSDLoadWithDescendantsLabel));
+    }
+
+    // If anything is loaded at this prim path or any of its descendants, add
+    // an item for unload.
+    if (!loadedSet.empty()) {
+        itemLabelPairs.emplace_back(std::make_pair(kUSDUnloadItem, kUSDUnloadLabel));
+    }
+
+    return itemLabelPairs;
+}
+
+//! \brief Get groups of concrete schema prim types to list dynamically in the UI
+static const std::vector<UsdUfe::SchemaTypeGroup>
+getConcretePrimTypes(bool sorted, const UsdContextOps::SchemaNameMap& schemaPluginNiceNames)
+{
+    std::vector<UsdUfe::SchemaTypeGroup> groups;
+
+    // Query all the available types
+    PlugRegistry&    plugReg = PlugRegistry::GetInstance();
+    std::set<TfType> schemaTypes;
+    plugReg.GetAllDerivedTypes<UsdSchemaBase>(&schemaTypes);
+
+    UsdSchemaRegistry& schemaReg = UsdSchemaRegistry::GetInstance();
+    for (auto t : schemaTypes) {
+        if (!schemaReg.IsConcrete(t)) {
+            continue;
+        }
+
+        auto plugin = plugReg.GetPluginForType(t);
+        if (!plugin) {
+            continue;
+        }
+
+        // For every plugin we check if there's a nice name registered and use that instead
+        auto plugin_name = plugin->GetName();
+        const auto search = schemaPluginNiceNames.find(plugin_name);
+        if (search != schemaPluginNiceNames.end())
+            plugin_name = search->second;
```

Comment:
New param uses here. This code used to have two std::vector<std::string> that it would find in first and then at same position in second. I changed it to a map.

Reply:
I dont know if its part of styling rules, but it seems everywhere else brackets are used with if statements

Reply:
Its not, but the surrounding code uses brackets, so I'll add them here.